### ISSUE TITLE
ctrl-c doesn't exit xmr-stak 2.10.3

### ISSUE THREAD

**Issue by lkelemen** [fireice-uk/xmr-stak#2388]
ctrl-c doesn't exit xmr-stak 2.10.3 on PC. It works on Mac

**Comment by todw1fd** [comment]
Having same issue running in a terminal on Win10.  Ctrl-c doesn't stop process correctly and drop to a prompt.  Same experience on three separate Win10 systems.

**Comment by Spudz76** [comment]
Weird I figured it was hanging on some GPU dumbness for me.  Win7.

Do you have Quickedit mode turned on of off?  Try it the other way idk.  Could be trying to kill+copy or something really dumb.  Try Ctrl+Break which is the original original old school Ctrl+C.

I thought I saw something change in the signal handling a little while ago but I don't remember when.  Something with the "proper" shutdown feature?

**Comment by psychocrypt** [comment]
I tested it on my windows test system (win10) and con not reproduce the issue. :-(

**Comment by todw1fd** [comment]
cmd, clicked multiple times but no drop to a prompt

**Comment by noebranco** [comment]
Same problem here but only with PC's using NVIDIA cards. All my other PC's close correctly. :-)

**Comment by Spudz76** [comment]
Yes, CUDA here too.  Was doing it on 10.1 for sure, but maybe not on 8.0 (I don't remember the 8.0 result).  All win7

**Comment by lkelemen** [comment]
my PC has Intel Core i7 7799 CPU, 32GB RAM,  win10 home version 1809 OS build 17763.404 (April 2, 2019—KB4490481), NVidia gtx 1080 driver version 25.21.14.1967 date 17.3.2019

**Comment by Ryo-RU** [comment]
Win 7. 
2.9.0 / 2.10.2 / 2.10.3 / 2.10.4 confirmed not closing using cuda10
**2.8.2** is closing ok

**Comment by bymaximus** [comment]
Same here guys, I noticied it since the DEV masari fix branch I was using, changed now to master, seems still have the problem. 

Ctrl+C doesnt terminate the process, it terminates the GPU workers because the GPU usage drops but doesnt terminate the main process. 

Im currently always killing the proccess with task manager, the process also doesnt accept other key commands also (like H).

Dll not named to cuda 10 but im using cuda 10.1.95, compiler showed it, driver 418.91

xmr-stak 2.10.4
[2019-04-12 17:00:18] : Mining coin: cryptonight_conceal
[2019-04-12 17:00:18] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10_0'
WARNING: NVIDIA cannot load backend library: xmrstak_cuda_backend_cuda10_0.dll
[2019-04-12 17:00:18] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_2'
WARNING: NVIDIA cannot load backend library: xmrstak_cuda_backend_cuda9_2.dll
[2019-04-12 17:00:18] : NVIDIA: try to load library 'xmrstak_cuda_backend'
NVIDIA: found 1 potential device's
[2019-04-12 17:00:18] : Starting NVIDIA GPU thread 0, no affinity.
CUDA [10.1/10.0] GPU#0, device architecture 52: "GeForce GTX 970"... device init succeeded
[2019-04-12 17:00:19] : NVIDIA: use library 'xmrstak_cuda_backend'
